### ISSUE TITLE
ScrollOverflow event order issue prevents to scroll to the bottom of a section

### ISSUE THREAD

**Issue by AllirionX** [alvarotrigo/fullPage.js#3527]
### Description
Using fullpage.js 2.9.7, I found an issue with Scrolloverflow. The issue can be reproduced on the demo page in 3.0.2. https://alvarotrigo.com/fullPage/examples/scrolling.html

When scrolling down inside a long section with a wheel, the last wheel event fires a section jump instead of simply scrolling inside of the current section. On the demo page, if you scroll down the first page you cannot read "Ea zril aliquip euismod sed." without jumping to the next section.

### Root cause
It is due to the check line 1068 of jquery.fullPage.js (2.9.7)
`options.scrollOverflowHandler.isScrolled(check, scrollable)`
 being performed after the wheel event is resolved in scrolloverflow.js. It results that the check to see if the section is scrolled is performed with the "future" scrolling coordinates, not the current one.

### Proposed fix
Make the wheel events attached to the fp-scrollable dom elements not capture the event in scrolloverflow.js
```
IScroll.prototype.wheelOn = function () 
    this.wrapper.addEventListener('wheel', this, false); /* Event is fired only in the bubbling phase */  
    this.wrapper.addEventListener('mousewheel', this, false);   /* Event is fired only in the bubbling phase */ 
    this.wrapper.addEventListener('DOMMouseScroll', this, false);  /* Event is fired only in the bubbling phase */ 
};
```

Make sure the wheel event attached to document are fired in fullpage.js
```
function addMouseWheelHandler(){
            var prefix = '';
            var _addEventListener;

            if (window.addEventListener){
                _addEventListener = "addEventListener";
            }else{
                _addEventListener = "attachEvent";
                prefix = 'on';
            }

             // detect available wheel event
            var support = 'onwheel' in document.createElement('div') ? 'wheel' : // Modern browsers support "wheel"
                      document.onmousewheel !== undefined ? 'mousewheel' : // Webkit and IE support at least "mousewheel"
                      'DOMMouseScroll'; // let's assume that remaining browsers are older Firefox


            if(support == 'DOMMouseScroll'){
                document[ _addEventListener ](prefix + 'MozMousePixelScroll', MouseWheelHandler, true); /* Event is fired in the capturing phase */
            }

            //handle MozMousePixelScroll in older Firefox
            else{
                document[ _addEventListener ](prefix + support, MouseWheelHandler, true); /* Event is fired in the capturing phase */
            }
        }```

**Comment by alvarotrigo** [comment]
Thanks for reporting it!
Please feel free to create a pull request with the proposed solution. I'll then review it.

**Comment by chrismou** [comment]
Ah thanks, I do apologise, that ticket didn't show up on a search. I'll check it out!

**Comment by alvarotrigo** [comment]
>When scrolling down inside a long section with a wheel, the last wheel event fires a section jump instead of simply scrolling inside of the current section. 

@AllirionX I asume you are talking about when you use a normal mouse wheel?
Not about kinetic scrolling devices such as Apple trackpads, some other laptops or Apple Magic mouses?

**Comment by AllirionX** [comment]
Yes, I am talking about normal mouse wheel.
You might want to recheck the proposed fix with other type of wheel event - I think I still had a problem with mobile device scroll, not 100% sure though.

**Comment by alvarotrigo** [comment]
The issue has been fixed in the dev branch! 💛 
It will get merged in the next release 3.0.6!

**Comment by alvarotrigo** [comment]
Merged into fullpage.js 3.0.6 which is now available! ⚡️ ⚡️
